### PR TITLE
Remove `check_includes = false` in `//brave/browser/tor/BUILD.gn`

### PR DESCRIPTION
**Pull request by sangwoo108**
The target was depending on `//chrome/browser` which caused cyclic dependency. By adding the target directly to `//chrome/browser` using `sources.gni` we can avoid the cyclic deps and `check_includes = false`.

There's no intended behavior change.

Resolves https://github.com/brave/brave-browser/issues/10655

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: browser/tor/sources.gni [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    "//third_party/blink/public/common",
+    "//url",
+  ]
+
+  # Below dep list are not directly used tor target.
+  # Only added to fix intermittent build failure.
+  # Most of the case comes from including brave_browser_process_impl.h`
+  # headers. That header includes some buildflags but sometimes they are
+  # not generated before evaluating tor target.
+  brave_browser_tor_deps += [
```
Comment (iefremov) [inline browser/tor/sources.gni:45]: @bridiver I think should be safe to drop these now when the code is in source.gni ?

File: browser/tor/sources.gni [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import("//brave/components/tor/buildflags/buildflags.gni")
+
+brave_browser_tor_sources = []
+brave_browser_tor_deps = []
+
+if (enable_tor) {
```
Comment (bridiver) [inline browser/tor/sources.gni:11]: please move tests in `browser/tor` to `browser/tor/test` so we don't have `BUILD.gn` and `source.gni` for the same directory

File: browser/tor/sources.gni (hunk 1 above)
Comment (bridiver) [reply browser/tor/sources.gni:45]: also  the include for brave_browser_process_impl.h in brave_vpn_button_unittest.cc is wrong and should be replaced with `brave_browser_process.h` and depend on `//brave/browesr:browser_process`

Comment (sangwoo108) [reply browser/tor/sources.gni]: brave_browser_process_impl.h seem to be unused in the file. Removed.

**Review by iefremov — APPROVED** [review]

**Pull request merged**